Prox/Prox/Models/Place.swift

```diff
@@ -84,62 +117,80 @@ class Place: Hashable {
-                  description: description,
+                  wikiDescription: wikiDescription,
+                  yelpDescription: yelpDescription,
                   latLong: CLLocationCoordinate2D(latitude: lat, longitude: lng),
                   categories: categoryNames,
                   url: value["url"] as? String,
                   address: (value["address"] as? [String])?.joined(separator: " "),
                   yelpProvider: ReviewProvider(fromFirebaseSnapshot: data.childSnapshot(forPath: YELP_PATH)),
                   tripAdvisorProvider: ReviewProvider(fromFirebaseSnapshot: data.childSnapshot(forPath: TRIP_ADVISOR_PATH)),
                   photoURLs: photoURLs,
-                  hours: nil)
+                  hours: hours)
     }
 
-
     static func ==(lhs: Place, rhs: Place) -> Bool {
         return lhs.id == rhs.id
     }
 
 }
 
-enum DayOfWeek: Int {
-    case monday = 0 // matches server representation
+enum DayOfWeek: String {
+    case monday
     case tuesday, wednesday, thursday, friday
     case saturday, sunday
 
-    static func forDate(_ date: Date) -> DayOfWeek {
-        let iOSWeekday = getiOSWeekday(forDate: date)
-        return weekday(fromiOSWeekday: iOSWeekday)
-    }
-
-    private static func getiOSWeekday(forDate date: Date) -> Int {
-        let cal = Calendar(identifier: .gregorian)
-        return cal.component(.weekday, from: date)
-    }
+    private static let Cal: Calendar = {
+        var cal = Calendar(identifier: .gregorian)
+        cal.locale = Locale(identifier: "en_US")  // necessary for Calendar.weekdaySymbols to return long symbols.
+        return cal
+    }()
 
-    private static func weekday(fromiOSWeekday iOSWeekday: Int) -> DayOfWeek {
-        // iOSWeekday is 1-7 where Sun = 1; we are 0-6 where Mon = 0
-        var dayInt = iOSWeekday - 2
-        if dayInt < 0 { // only Sunday
-            dayInt = 6
-        }
-        return DayOfWeek(rawValue: dayInt)!
+    static func forDate(_ date: Date) -> DayOfWeek {
+        let weekdayInt = Cal.component(.weekday, from: date)
+        let weekdayStr = Cal.weekdaySymbols[weekdayInt - 1]
+        return DayOfWeek(rawValue: weekdayStr.lowercased())!
     }
 }
 
 struct OpenHours {
-    let startTime: Int
-    let endTime: Int
+
+    private static let TimeInDay: TimeInterval = 60 * 60 * 24 // specified in seconds.
+
+    /* Notes:
+     *   - An entry for DayOfWeek is missing if a location is not open that day.
+     *   - For simplicity, we just use the last open interval for a day.
+     *   - For simplicity, dates are normalized by a single day, allowing comparison but preventing
+     *     the stored dates from being correct (e.g. if today is Mon Nov 7, the stored date for Tues
+     *     isn't Nov 8).
+     *
+     * TODO:
+     *   - Use all open intervals
+     *   - Use accurate days of week in Date (maybe? It adds complexity for little gain).
+     */
+    let hours: [DayOfWeek : (open: Date, close: Date)]
```

Comment:
`open` is now a reserved word in Swift 3.0 (like `private` and `public`. I'd be more comfortable if we could find a different name as I am uncertain what the result of this might be long term.